tests/Backend/Snowflake/OrderByTest.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\Test\Backend\Snowflake;
+
+use Keboola\Csv\CsvFile;
+use Keboola\StorageApi\Client;
+use Keboola\StorageApi\ClientException;
+use Keboola\StorageApi\TableExporter;
+use Keboola\Test\StorageApiTestCase;
+
+class OrderByTest extends StorageApiTestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->_initEmptyTestBuckets();
+    }
+
+    public function testSimpleSort(): void
+    {
+        $tableId = $this->prepareTable();
+
+        $order = [
+            'column' => 'column_string',
+        ];
+
+        $dataPreview = $this->_client->getTableDataPreview($tableId, ['orderBy' => [$order]]);
+        $exportTable = $this->getExportedTable($tableId, ['orderBy' => [$order]]);
```

Comment:
Ještě jsem se díval jestli to vyexportuje více souborů aby šlo vidět že to správně řadí viz. https://support.snowflake.net/s/case/5000Z000010uLgIQAU/unloading-sorted-csv?tabset-5ba45500=2

> Yes. If the files are concatenated in the sequential order, then it'll be correctly sorted.

Stahuje to v pořadí v jakém jsou vloženy ty soubory do manifestu https://github.com/keboola/storage-api-php-client/blob/c49dea9c373cb449178d714a2c58758e99bfd187/src/Keboola/StorageApi/TableExporter.php#L89

A do manifestu se zas vkládají tak jak to vylistuje S3 https://github.com/keboola/connection/blob/9604c3bf61cb010ecf4ef096fba0a8aa5db3399c/application/modules/storage/services/BucketBackend/Snowflake.php#L426 a ta to mám listovat v alfabetickém pořadí takže bych řekl že ok https://docs.aws.amazon.com/aws-sdk-php/v3/api/api-s3-2006-03-01.html#listobjects

Reply:
V testu to ale exportuje do jednoho souboru a asi by se tam muselo namlátit hodně dat aby to slicnulo. Takže bych možná jenom vyzkoušel jednorázově.

Reply:
Snížil jsem `MAX_FILE_SIZE` na 5000 a zkusil udělat export. Udělalo se asi 16 slices
```
    public function testSlices()
    {
        $csvFile = new CsvFile(tempnam(sys_get_temp_dir(), 'keboola'));
        $csvFile->writeRow(['id', 'data']);
        for ($i = 0; $i++ < 100000;) {
            $csvFile->writeRow([$i, $this->randomString(1000)]);
        }
        $tableId = $this->_client->createTableAsync($this->getTestBucketId(), 'conditions', $csvFile);

        $sort = [
            [
                'column' => 'id',
                'order' => 'DESC',
                'dataType' => 'INTEGER'
            ]
        ];

        $table = $this->getExportedTable($tableId, ['orderBy' => $sort]);
        $last = null;
        foreach ($table as $row) {
            if ($last) {
                $this->assertEquals($row['id'] + 1, $last['id']);
            }
            $last = $row;
        }
    }
```
 a všechno bylo správně sesortovaný, takže :+1:

Reply:
paráda 👍

Reply:
![snimek z 2019-02-27 14-35-53](https://user-images.githubusercontent.com/5272798/53494017-0cc4a080-3a9d-11e9-94a3-1b679cff4ffc.png)

Reply:
ten test nikam pushovat nebudu, protože trvá 5 minut a smažu ho